service/common/src/main/java/org/apache/polaris/service/catalog/policy/PolicyCatalog.java

```diff
@@ -525,19 +540,91 @@ private static Policy constructPolicy(PolicyEntity policyEntity) {
         .build();
   }
 
-  private static ApplicablePolicy constructApplicablePolicy(
-      PolicyEntity policyEntity, boolean inherited) {
+  private boolean filterApplicablePolicy(PolicyEntity policyEntity) {
+    // check the type
+    if (policyEntity.getPolicyType().equals(ACCESS_CONTROL)) {
```

Comment:
Need a PolicyEntityAugmentor which could conditionally modify the object, working on it